src/c/soubory/otevirani_souboru.md

```diff
@@ -0,0 +1,157 @@
+# Otevírání souborů
+Abychom mohli s nějakým souborem začít pracovat, musíme ho nejprve v našem programu otevřít, aby
+byl vytvořen souborový deskriptor, do kterého pak můžeme zapisovat či z něho číst data. K tomu slouží
+funkce [`fopen`](https://devdocs.io/c/io/fopen), která má následující
+[signaturu](../funkce/funkce.md#syntaxe):
+```c
+FILE* fopen(const char* filename, const char* mode);
+```
+
+## Cesta k souboru
+Jako svůj první parametr funkce `fopen` očekává řetězec s cestou k souboru, který má být otevřen.
+Cestu můžete zadat dvěma způsoby:
+- **Absolutní cesta** (*absolute path*) je cesta, která začíná kořenovým adresářem souborového
+systému, například `/home/student/upr/soubor.txt`[^1]. Aby byla cesta absolutní, musí na Linuxu
+začínat dopředným lomítkem.
+- **Relativní cesta** (*relative path*) se vyhodnotí relativně k tzv. **pracovnímu adresáři**
+(*working directory*) běžícího programu. Pokud spustíte váš program z terminálu, tak se pracovní
+adresář implicitně nastaví na adresář, ze kterého jste program spustili. Pokud tedy například spustíte
+váš program ze složky `/home/student/upr` a funkci `fopen` předáte cestu `soubor.txt`, tak se funkce
+pokusí otevřít soubor na cestě `/home/student/upr/soubor.txt`.
+
+Při zadávání cesty můžete využít zkratky `.` a `..`, které jsou užitečné zejména u relativních cest:
+- Zkratka `.` se odkazuje na současný adresář, `./soubor.txt` je tedy to samé jako `soubor.txt`.
+- Zkratka `..` se odkazuje na rodičovský adresář, `../data/abc.txt` tedy říká:
+`Podívej se do rodičovské složky, tam nalezni adresář data a v něm soubor abc.txt`.
+
+Nepokoušejte se však zadávat cesty k neexistujícím složkám. `fopen` sice vytvoří nový soubor, pokud
+neexistuje, ale neexistující složku za vás nevytvoří.
+
+> Doposud jsme používali prvky *C*, které byly vesměs nezávislé na použitém operačním systému. Jakmile
+> ale naše programy začnou interagovat se **souborovým systémem** (*file system*), budeme muset začít
+> respektovat zákonitosti operačního systému, na kterém náš program poběží. Proto například u cesty
+> k souborům vždy používejte dopředná lomítka (`/`) pro oddělování adresářů, pokud program budete
+> spouštět na Linuxu. 
+
+[^1]: Na Windows by podobná cesta mohla vypadat například jako
+`C:\Users\student\upr\soubor.txt`.
```

Comment:
Pokud mne pamet neklame, tak by mohlo fungovat i `C://Users//student//upr//soubor.txt`

Reply:
Windows zvládne i normální dopředná lomítka (tj. `/`, ne `//`), ale chtěl jsem to dát jako příklad rozdílu.